### DISCUSSION TITLE
`createClient` best practices in serverless functions

### DISCUSSION THREAD

**Discussion by ryanleecode** [polkadot-api/polkadot-api#653] [Q&A]
Is it recommended or not recommend to put `createClient` outside of a serverless function `handler` function such that it can be reused across invokations? is there any downside to this?

for example...

```ts
const client = createClient(getWsProvider(RPC_ENDPOINT));

export const handler = async () => {
   ...
}

```

vs

```ts
export const handler = async () => {
  const client = createClient(getWsProvider(RPC_ENDPOINT));
 try {
   ...
 } finally {
   client.destroy()
 }
}
```

**Comment by josepot** [comment]
It all depends on the kind of serverless function environment that you are working on with... Please ask your service provider.

That being said, this library is not optimized for serverless function (and neither is PJS for that matter)
